### PR TITLE
added `deleteDataset($datasetId)` to destroy datasets

### PR DESCRIPTION
**Pull request by joenio**
## Description

This new method make a request to the following address:

* DELETE /api/datasets/:persistentId/destroy/?persistentId=$datasetId

This Dataverse API endpoint let superusers to remove published datasets and
ordinary users to remove draft/unpublished datasets as can see on the API spec
docs:

* https://guides.dataverse.org/en/latest/api/native-api.html#delete-published-dataset

## Changes
Added new method: `deleteDataset($datasetId)`.

## Tests

1. Create a new dataset, keep it unpublished/draft
1. Call the new method `deleteDataset($datasetId)` passing the persistendId of the dataset on $datasetId param
1. The dataset should be destroyed

## Checklist

- [ ] The project builds
- [ ] The project passes lint checks
- [ ] The project passes format checks
- [ ] The project passes unit tests
- [ ] I've manually tested the functionality

## Screenshots

Captura de tela de 2020-11-02 15-38-19

Captura de tela de 2020-11-02 15-38-33

### CODE REVIEW

**Review by tainguyenbui — APPROVED** [review]
LGTM

**Comment by tainguyenbui** [comment]
@joenio Once this is merged, we would need to deploy a new version of the library, so please give me a thumbs up if you need it asap and I will take in charge of doing so

**Comment by joenio** [comment]
@tainguyenbui thanks for your quickly feedback here, I'm going to start work on the issue #30 also, maybe it's a good idea to wait it to release a new package including this also, but I don't know how much time it takes me as I'm not much experienced with Dataverse API yet.

**Comment by tainguyenbui** [comment]
> @tainguyenbui thanks for your quickly feedback here, I'm going to start work on the issue #30 also, maybe it's a good idea to wait it to release a new package including this also, but I don't know how much time it takes me as I'm not much experienced with Dataverse API yet.

No worries! you can continue adding changes to this PR. I would encourage adding something in the PR title such as `[WIP]` or `[DO NOT MERGE]`. Thanks for contributing 😊

**Comment by joenio** [comment]
> > @tainguyenbui thanks for your quickly feedback here, I'm going to start work on the issue #30 also, maybe it's a good idea to wait it to release a new package including this also, but I don't know how much time it takes me as I'm not much experienced with Dataverse API yet.
> 
> No worries! you can continue adding changes to this PR. I would encourage adding something in the PR title such as `[WIP]` or `[DO NOT MERGE]`. Thanks for contributing blush

Thanks, but I will open new contributions in new PR, this PR here is finished. BTW, I like to know about this practice of putting info on the PR title.

**Pull request merged**
